From 930e32081c76c42f4249f404d06dd471d333d1e2 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 17 Mar 2013 10:36:16 +0100
Subject: [PATCH] opencv pkgconfig: fix cmake entries to abstract $libdir

The current pkgconfig leads to missing symbols and RDEPENDS in e.g. gst-plugins-bad-opencv

Patch downloaded from http://pkgs.fedoraproject.org/cgit/opencv.git/tree/opencv-pkgcmake.patch
 and http://pkgs.fedoraproject.org/cgit/opencv.git/tree/opencv-pkgcmake2.patch on 20130316

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>

Upstream-Status: Unknown
http://code.opencv.org/issues/1925
---
 cmake/OpenCVGenPkgconfig.cmake        | 10 ++++++----
 cmake/templates/OpenCVConfig.cmake.in |  6 +-----
 2 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/cmake/OpenCVGenPkgconfig.cmake b/cmake/OpenCVGenPkgconfig.cmake
index 49d6707..e087f66 100644
--- a/cmake/OpenCVGenPkgconfig.cmake
+++ b/cmake/OpenCVGenPkgconfig.cmake
@@ -10,7 +10,7 @@
 # -------------------------------------------------------------------------------------------
 set(prefix      "${CMAKE_INSTALL_PREFIX}")
 set(exec_prefix "\${prefix}")
-set(libdir      "") #TODO: need link paths for OpenCV_EXTRA_COMPONENTS
+set(libdir      "\${prefix}/${OPENCV_LIB_INSTALL_PATH}")
 set(includedir  "\${prefix}/${OPENCV_INCLUDE_INSTALL_PATH}")
 set(VERSION     ${OPENCV_VERSION})
 
@@ -36,7 +36,7 @@ ocv_list_reverse(OpenCV_LIB_COMPONENTS)
 ocv_list_reverse(OpenCV_EXTRA_COMPONENTS)
 
 #build the list of components
-set(OpenCV_LIB_COMPONENTS_ "")
+set(OpenCV_LIB_COMPONENTS_ "-L\${libdir}")
 foreach(CVLib ${OpenCV_LIB_COMPONENTS})
   get_target_property(libpath ${CVLib} LOCATION_${CMAKE_BUILD_TYPE})
   get_filename_component(libname "${libpath}" NAME)
@@ -51,8 +51,10 @@ foreach(CVLib ${OpenCV_LIB_COMPONENTS})
   else()
     set(installDir "${OPENCV_LIB_INSTALL_PATH}")
   endif()
-
-  set(OpenCV_LIB_COMPONENTS_ "${OpenCV_LIB_COMPONENTS_} \${exec_prefix}/${installDir}/${libname}")
+  string(REPLACE "libopencv" "-lopencv" libname "${libname}")
+  string(REPLACE ".so"    "" libname "${libname}")
+  string(REPLACE ".dylib" "" libname "${libname}")
+  set(OpenCV_LIB_COMPONENTS_ "${OpenCV_LIB_COMPONENTS_} ${libname}")
 endforeach()
 
 # add extra dependencies required for OpenCV
diff --git a/cmake/templates/OpenCVConfig.cmake.in b/cmake/templates/OpenCVConfig.cmake.in
index 4ff3569..64199a1 100644
--- a/cmake/templates/OpenCVConfig.cmake.in
+++ b/cmake/templates/OpenCVConfig.cmake.in
@@ -53,11 +53,7 @@ set(OpenCV_USE_MANGLED_PATHS @OpenCV_USE_MANGLED_PATHS_CONFIGCMAKE@)
 get_filename_component(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_FILE}" PATH CACHE)
 
 if(NOT WIN32 OR OpenCV_ANDROID_NATIVE_API_LEVEL GREATER 0)
-  if(OpenCV_ANDROID_NATIVE_API_LEVEL GREATER 0)
-    set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../..")
-  else()
-    set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../..")
-  endif()
+  set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../..")
   # Get the absolute path with no ../.. relative marks, to eliminate implicit linker warnings
   if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_LESS 2.8)
     get_filename_component(OpenCV_INSTALL_PATH "${OpenCV_INSTALL_PATH}" ABSOLUTE)
-- 
1.8.1.4

